src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -0,0 +1,223 @@
+            }
+
+            int longestMatch = -1;
+
+            // Pick the ctor with the longest parameter list where all parameters are matched. 
+            var ctors = t.GetConstructors(BindingFlags.Public | BindingFlags.Instance);
+            foreach (var ctor in ctors)
+            {
+                var ps = ctor.GetParameters();
+                int len = ps.Length;
+
+                var getArgFuncs = new Func<IReadOnlyDictionary<string, object>, object>[len];
+
+                bool ok = true;
+                for (int i = 0; i < len; i++)
+                {
+                    var p = ps[i];
+                    PropertyInfo propInfo = null;
+                    if (!availableParams.TryGetValue(p.Name, out propInfo))
+                    {
+                        ok = false;
+                        break;
+                    }
+                    bool resolve = propInfo.GetCustomAttribute<ResolvePropertyAttribute>() != null;
+                    getArgFuncs[i] = (bindingData) =>
+                    {
+                        var propValue = propInfo.GetValue(_source);
+                        if (resolve)
+                        {
+                            string str = (string)propValue;
+                            if (str != null)
+                            {
+                                str = _nameResolver.ResolveWholeString(str);
+
+                                BindingTemplate template = BindingTemplate.FromString(str);
+                                propValue = template.Bind(bindingData);
+                            }
+                        }
+                        return propValue;
+                    };
+                }
+
+                if (ok)
+                {
+                    if (len > longestMatch)
+                    {
+                        _bestCtor = ctor;
+                        longestMatch = len;
+                        _bestCtorArgBuilder = getArgFuncs;
+                    }
+                }
+            }
+
+            if (_bestCtor == null)
+            {
+                // error!!!
+                throw new InvalidOperationException("Can't figure out which ctor to call.");
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1822:MarkMembersAsStatic")]
+        public string GetInvokeString(TAttribute attributeResolved)
+        {
+            var resolver = GetAttributeResolve(attributeResolved);
+            string invokeString = resolver.ToInvokeString();
+            return invokeString;
+        }
+
+        public TAttribute ResolveFromInvokeString(string invokeString)
+        {
+            IAttributeInvokeDescriptor<TAttribute> resolver = GetAttributeResolve(_source);
+            var attr = resolver.FromInvokeString(invokeString);
+            return attr;
+        }
+
+        
+        private static IAttributeInvokeDescriptor<TAttribute> GetAttributeResolve(TAttribute attr)
```

Comment:
This method doesn't seem named right. Perhaps call it GetInvokeDescriptor or something?